core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -91,7 +90,12 @@ private[spark] class DiskBlockManager(blockManager: BlockManager, conf: SparkCon
   /** List all the files currently stored on disk by the disk manager. */
   def getAllFiles(): Seq[File] = {
     // Get all the files inside the array of array of directories
-    subDirs.flatten.filter(_ != null).flatMap { dir =>
+    subDirs.flatMap { dir =>
```

Comment:
The old implementation didn't access the `Array`s in `dir.synchronized`. So even if we read a non-null file from these `Array`s, we still don't have any happen before relation about creating the file object and reading the non-null value of the file pointer, and of cause `mkdir` is also same.